### ISSUE TITLE
getContent() method returns wrong content part

### ISSUE THREAD

**Issue by billynoah** [ddeboer/imap#342]
If a message has an "eml" attachment with attachments of it's own the `getContent()` method can return the wrong content part - resulting in the issue described here: https://github.com/ddeboer/imap/issues/284

For example the `getDecodedContent()` method attempts to decode a base64 encoded image and when it calls `$this->getContent()` the 'text/html' portion of the email might be returned.  I am sure this has to do with the fact that the attachments themselves can contain a "boundary" thereby confusing the parser, but at the moment a workaround is a bit beyond me.

**Comment by Slamdunk** [comment]
You can find my email in commit patches like: https://github.com/ddeboer/imap/commit/888139dd42b0b0ed06493168365ea3b1f762abdc.patch

**Comment by billynoah** [comment]
You should be able top replicate the issue with the following snippet.  Iterating over the messages, when I get to the jpg image and run `$attachment->getDecodedContent()` I end up with the `text/html` content of the message.

```
$attachments = $message->getAttachments();
foreach ($attachments as $attachment) {
    try {
        file_put_contents(
            '/write/path/for/attachment/' . $attachment->getFilename(),
            $attachment->getDecodedContent()
        );
    } catch(Exception $e) {
        die($e->getMessage());
    }
}
```

**Comment by Slamdunk** [comment]
I've been able to reproduce the bug in #343 onto a dataset with fake content. Thank you for reporting it: I hope to be able to fix it soon.

**Comment by Slamdunk** [comment]
https://secure.php.net/manual/en/function.imap-fetchbody.php#89002

**Comment by Slamdunk** [comment]
Bugfix released in [1.5.1](https://github.com/ddeboer/imap/releases/tag/1.5.1)
